tasks/main.yml

```diff
@@ -45,10 +45,21 @@
     regexp: SSH_AUTH_SOCK
     line: Defaults env_keep += "SSH_AUTH_SOCK"
 
-- name: install base packages
+- name: install 16.04-specific base packages
   apt: pkg={{ item }} state=present update-cache=yes cache_valid_time=3600
+  when: ansible_distribution == 'Ubuntu' and ansible_distribution_version == '16.04'
   with_items:
     - python-software-properties
+
+- name: install 18.04-specific base packages
+  apt: pkg={{ item }} state=present update-cache=yes cache_valid_time=3600
+  when: ansible_distribution == 'Ubuntu' and ansible_distribution_version == '18.04'
```

Comment:
And this will probably be needed for 18.04 and up.

I think something like this would work:

`when: ansible_distribution == 'Ubuntu' and ansible_lsb["ansible_major_release"]|int >= 18`

(untested)

Reply:
Updated using the example here and tested on my deploy: https://docs.ansible.com/ansible/2.5/user_guide/playbooks_conditionals.html#the-when-statement